generate-text-packs.sh

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
+
+set -e
+
+usage()
+{
+	echo "Usage: $0 --name <package_name> --versions <package_versions> [options]"
+	echo ""
+	echo "  Restores specified package versions and generates a source-build-usable text-only package."
+	echo "  The resulting text-only packages can be found in the './src/textOnlyPackages/src' directory."
+	echo ""
+	echo "  Both --name and --versions must be specified"
+	echo ""
+	echo "  --name <package_name>          Name of the package. Example: Microsoft.Docker.Sdk (short: -n)"
+	echo "  --versions <package_versions>  Versions of the package, separated by a comma. Example: 1.1.0,1.0.0 (short: -v)"
+	echo ""
+	echo "Options:"
+	echo "  --feeds \"<value>\"            Semicolon-separated list of additional Nuget feeds to use during restore (short: -f)"
+}
+
+source="${BASH_SOURCE[0]}"
+
+# resolve $SOURCE until the file is no longer a symlink
+while [[ -h $source ]]; do
+  scriptroot="$( cd -P "$( dirname "$source" )" && pwd )"
+  source="$(readlink "$source")"
+
+  # if $source was a relative symlink, we need to resolve it relative to the path where the
+  # symlink file was located
+  [[ $source != /* ]] && source="$scriptroot/$source"
+done
+
+scriptroot="$( cd -P "$( dirname "$source" )" && pwd )"
+
+package_name=''
+package_versions=''
+feeds=''
+
+if [[ $# -le 0 ]]; then
+	usage
+	exit 0
+fi
+
+while [[ $# > 0 ]]; do
+	opt="$(echo "${1/#--/-}" | tr "[:upper:]" "[:lower:]")"
+	case "$opt" in
+		-help|-h)
+			usage
+			exit 0
+			;;
+		-name|-n)
+			package_name=$2
+			shift
+			;;
+		-versions|-v)
+			package_versions=$2
+			shift
+			;;
+		-feeds|-f)
+			feeds=$2
+			shift
+			;;
+		*)
+			echo -e "ERROR: Unknown option: $1"
+			echo ""
+			usage
+			exit 1
+			;;
+	esac
+
+	shift
+done
+
+if [ -z "$package_name" ] || [ -z "$package_versions" ]; then
+	echo -e "ERROR: Both package name and version need to be specified!"
+	exit 1
+fi
+
+# -p:RestoreAdditionalProjectSources -> specifies additional Nuget package sources, including feeds: https://docs.microsoft.com/en-us/nuget/reference/msbuild-targets#restore-properties
+# -p:RestoreUsingNuGetTargets -> switch that controls who will restore the project; if set to `false`, Arcade's Build.proj will execute the Restore target of the project itself.
```

Comment:
We could, theoretically, also set the `RestoreUsingNuGetTargets` property inside of the `.proj` file and this way we embed this silent contract of how the proj file needs to be run inside of it. Would that work?

Something like
```xml
<RestoreUsingNuGetTargets Condition=" '$(RestoreUsingNuGetTargets)' != '' ">false</RestoreUsingNuGetTargets>
```

Reply:
Someone can brake it by accident by explicitly setting `/p:RestoreUsingNuGetTargets=true`. Furthermore, this option is only needed if you utilize Arcade's `build/sh`. If someone were to run the project via `MSBuild` directly, it would work without the property, making the assignment inside the `.proj` file redundant

Since we only have 1 entry-point and the property is specific to this entry-point and not to the project itself, I would keep it here

Reply:
Sure, if we always keep the `.proj` with the `.sh` it makes sense so hopefully they will.